paimon-core/src/test/java/org/apache/paimon/TestFileStore.java

```diff
@@ -363,7 +363,9 @@ public List<Snapshot> commitDataImpl(
 
         List<Snapshot> snapshots = new ArrayList<>();
         for (long id = snapshotIdBeforeCommit + 1; id <= snapshotIdAfterCommit; id++) {
-            snapshots.add(snapshotManager.snapshot(id));
+            if (snapshotManager.snapshotExists(id)) {
+                snapshots.add(snapshotManager.snapshot(id));
+            }
```

Comment:
If a dirty commit occurs, the previous processing logic throws an exception, which affects the running of the test case.